query_execution.go

```diff
@@ -149,10 +149,13 @@ func (q *queryExecution) writeExecutionResult(step *QueryPlanStep, data interfac
 }
 
 func (q *queryExecution) executeChildStep(step *QueryPlanStep, boundaryIDs []string) error {
-	atomic.AddInt32(&q.requestCount, 1)
```

Comment:
You could get rid of the unlock in both branches with something like:

```
func (q *queryExecution) incrementAndReturnRequestCount() int32 {
    q.mutex.Lock()
    defer q.mutex.Unlock()
    q.requestCount += 1
    return q.requestCount
}
```

Reply:
You could but you'd make the entire func into a critical section and all steps would run sequentially instead of in parallel :)